### PR TITLE
Set covering

### PR DESCRIPTION
**Pull request by c-allergic**

### CODE REVIEW

File: test/models/SetCovering.jl [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @test flavors(SetCovering) == [0, 1]
+    
+    # weights interface
+    @test parameters(c) == [1, 1, 2]
+    @test set_parameters(c, [1, 2, 3]) == SetCovering([[1, 2], [2, 3], [2, 3, 4]], [1, 2, 3])
+    
+    # evaluate
+    @test evaluate(c, [0, 1, 1]) == 3
+    @test is_set_covering(c,[1,0,1]) == true
+    @test is_set_covering(c,[0,0,1]) == false
```
Comment (GiggleLiu) [inline test/models/SetCovering.jl:18]: Please also test `findbest` function.

File: src/models/SetCovering.jl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        new{ET, typeof(weights)}(sets, weights)
+    end
+end
+Base.:(==)(a::SetCovering, b::SetCovering) = a.sets == b.sets && a.weights == b.weights
+
+#variables interface
+variables(gp::SetCovering) = gp.sets
+num_variables(c::SetCovering) = length(c.sets)
+flavors(::Type{<:SetCovering}) = [0, 1] #whether the set is selected or not
+# num_flavors(::Type{<:SetCovering}) = 2, here the number of flavors is always 2
```
Comment (GiggleLiu) [inline src/models/SetCovering.jl:27]: Clean up.

**Review by GiggleLiu — APPROVED** [review]
Good job!

File: test/models/SetCovering.jl (hunk 1 above)
Comment (c-allergic) [reply test/models/SetCovering.jl:18]: The src file didn't include a ```findbest``` function

**Comment by c-allergic** [comment]
- I add the ```findbest``` function and I found that I didn't properly use the ```BruteForce``` defined in src/bruteforce.jl. How could I fix it? 
- I think that maybe I should add the markdown file for these models not already included but I'm afraid that we would come back to add or change something about the interfaces.

**Pull request merged**